config/unmarshal_test.go

```diff
@@ -161,19 +161,17 @@ func TestConfigFromEnviroVars(t *testing.T) {
 		assert.Equal(t, "From Environment", *cfg.TeamSettings.SiteName)
 		assert.Equal(t, "Custom Brand", *cfg.TeamSettings.CustomBrandText)
 
-		if teamSettings, ok := envCfg["TeamSettings"]; !ok {
-			require.Fail(t, "TeamSettings is missing from envConfig")
-		} else if teamSettingsAsMap, ok := teamSettings.(map[string]interface{}); !ok {
-			require.Fail(t, "TeamSettings is not a map in envConfig")
-		} else {
-			if siteNameInEnv, ok := teamSettingsAsMap["SiteName"].(bool); !ok || !siteNameInEnv {
-				require.Fail(t, "SiteName should be in envConfig")
-			}
+		teamSettings, ok := envCfg["TeamSettings"]
+		require.True(t, ok, "TeamSettings is missing from envConfig")
 
-			if customBrandTextInEnv, ok := teamSettingsAsMap["CustomBrandText"].(bool); !ok || !customBrandTextInEnv {
-				require.Fail(t, "SiteName should be in envConfig")
-			}
-		}
+		teamSettingsAsMap, ok := teamSettings.(map[string]interface{})
+		require.True(t, ok, "TeamSettings is not a map in envConfig")
+
+		siteNameInEnv, ok := teamSettingsAsMap["SiteName"].(bool)
+		require.True(t, ok || siteNameInEnv, "SiteName should be in envConfig")
```

Comment:
shouldn't it be `ok && siteNameInEnv`?

Reply:
The original code reads:

```
if siteNameInEnv, ok := teamSettingsAsMap["SiteName"].(bool); !ok || !siteNameInEnv {
				t.Fatal("SiteName should be in envConfig")
			}
```

This would seem to indicate the test should fail if either `ok` is False/Not, _or_ siteNameInEnv is False/Not. Wouldn't `&&` be a change in the original boolean logic, requiring both conditions to be false to return a test failure?

Reply:
apologies, I misread the code :)